src/test/ui/issues/issue-46843.stderr

```diff
@@ -8,6 +8,9 @@ LL | |     Thing::This => 1,
 LL | |     Thing::That => 0
 LL | | };
    | |_^
+   |
+   = note: branching and conditionals are experimental in a const context
+   = help: add `#![feature(const_if_match)]` to the crate attributes to enable
```

Comment:
Let's enable the gate here (conditionally) and move the test into `control-flow/` ?